src/app/screen-snippet-handler.ts

```diff
@@ -16,187 +23,279 @@ import { windowExists } from './window-utils';
-        try {
-            if (!this.outputFileName) {
-                logger.info(`screen-snippet-handler: screen capture failed! output file doesn't exist!`);
-                return { message: 'output file name is required', type: 'ERROR' };
-            }
-            const data = await readFile(this.outputFileName);
-            if (!data) {
-                logger.info(`screen-snippet-handler: screen capture failed! data doesn't exist!`);
-                return { message: `no file data provided`, type: 'ERROR' };
-            }
-            // convert binary data to base64 encoded string
-            const output = Buffer.from(data).toString('base64');
-            return { message: 'success', data: output, type: 'image/png;base64' };
-        } catch (error) {
-            // no such file exists or user likely aborted
-            // creating snippet. also include any error when
-            // creating child process.
-            return error && error.code === 'ENOENT'
-                ? { message: `file does not exist`, type: 'ERROR' }
-                : { message: `${error}`, type: 'ERROR' };
-        } finally {
-            if (this.focusedWindow && windowExists(this.focusedWindow)) {
-                this.focusedWindow.moveTop();
-            }
-            // remove tmp file (async)
-            if (this.outputFileName) {
-                fs.unlink(this.outputFileName, (removeErr) => {
-                    logger.info(`screen-snippet-handler: cleaning up temp snippet file: ${this.outputFileName}!`);
-                    if (removeErr) {
-                        logger.error(`screen-snippet-handler: error removing temp snippet file: ${this.outputFileName}, err: ${removeErr}`);
-                    }
-                });
-            }
-        }
+  /**
+   * Verify and updates always on top
+   */
+  private async verifyAndUpdateAlwaysOnTop(): Promise<void> {
+    if (this.shouldUpdateAlwaysOnTop) {
+      await updateAlwaysOnTop(true, false, false);
+      this.shouldUpdateAlwaysOnTop = false;
     }
+  }
 
-    /**
-     * Verify and updates always on top
-     */
-    private async verifyAndUpdateAlwaysOnTop(): Promise<void> {
-        if (this.shouldUpdateAlwaysOnTop) {
-            await updateAlwaysOnTop(true, false, false);
-            this.shouldUpdateAlwaysOnTop = false;
-        }
+  /**
+   * Gets the height & width of an image
+   */
+  private getImageSize():
+    | {
+        height: number | undefined;
+        width: number | undefined;
+      }
+    | undefined {
+    if (!this.outputFileName) {
+      return undefined;
     }
+
+    const dimensions = sizeOf(this.outputFileName);
+    return {
+      height: dimensions.height,
+      width: dimensions.width,
+    };
+  }
+
+  /**
+   * Uploads a screen snippet
+   * @param webContents A browser window's web contents object
+   */
+  private uploadSnippet(webContents: Electron.webContents) {
+    ipcMain.on('upload-snippet', async (_event, snipImage: string) => {
```

Comment:
I see that this method is called every time `capture` is invoked. so, I think maybe we should we `ipcMain.once` to prevent memory leak?